automation/features/step-definitions/staking.steps.js

```diff
@@ -1,18 +1,66 @@
 const StakingPage = require("../pageobjects/staking.page");
 let currentAmountOfTokensInWallet =''
-
-Then(/^I can see the validator node list$/, () => {
-  const nodeList = $(".node-list").$$('[data-testid="node-list-item"]');
-  expect(nodeList.length).toBeGreaterThan(0);
-});
+let validatorName =''
 
 When(/^I associate some tokens from wallet$/,()=>{
     browser.getByTestId('callout').waitForDisplayed({timeout:30000,reverse:false,timeoutMsg: "callout was not found"})
+    browser.pause(2000)
   if (browser.getByTestId('associate-more-tokens-btn').isDisplayed()){
     browser.getByTestId('associate-more-tokens-btn').click()
   } else browser.getByTestId('associate-tokens-btn').click()
-  $('[data-testid="associate-radio-wallet"]').click({force:true})
+  browser.getByTestId('associate-radio-wallet').click({force:true})
+    browser.pause(1000)
+    browser.getByTestId('token-amount-input').setValue(0.00001)
+    browser.pause(1000)
+    browser.getByTestId('token-input-submit-button').click()
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
+    $('button=Confirm').click()
+    browser.switchWindow("VEGA");
+    browser.pause(2000)
+})
+
+When(/^I associate some tokens from vesting contract$/,()=>{
+    browser.getByTestId('callout').waitForDisplayed({timeout:30000,reverse:false,timeoutMsg: "callout was not found"})
+  if (browser.getByTestId('associate-more-tokens-btn').isDisplayed()){
+    browser.getByTestId('associate-more-tokens-btn').click()
+  } else browser.getByTestId('associate-tokens-btn').click()
+  browser.getByTestId('associate-radio-contract').click({force:true})
+    browser.pause(1000)
+    browser.getByTestId('token-amount-input').setValue(0.00001)
+    browser.pause(1000)
+    browser.getByTestId('token-input-submit-button').click()
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
+    $('button=Confirm').click()
+    browser.switchWindow("VEGA");
+    browser.pause(2000)
+})
+
+When(/^I disassociate some tokens from wallet$/,()=>{
+    browser.getByTestId('callout').waitForDisplayed({timeout:30000,reverse:false,timeoutMsg: "callout was not found"})
+    browser.getByTestId("disassociate-tokens-btn").waitForDisplayed({timeout:30000,reverse:false,timeoutMsg: "disassociate btn  was not found"})
+    browser.getByTestId("disassociate-tokens-btn").click()
+    browser.pause(1000)
+    $('[data-testid="associate-radio-wallet"]').click({force:true})
+    browser.pause(1000)
+    browser.getByTestId('token-amount-input').setValue(0.00001)
+    browser.pause(1000)
+    browser.getByTestId('token-input-submit-button').click()
+    browser.waitUntil(() => browser.getWindowHandles().length > 1);
+    browser.switchWindow("MetaMask Notification");
+    $('button=Confirm').click()
+    browser.switchWindow("VEGA");
+    browser.pause(2000)
```

Comment:
There's a lot of test ids here that should be added to the page object. There's also a lot of implicit waits here, is it possible to do something like wait for element to be displayed?